lab_4_retrieval_w_clustering/main.py

```diff
@@ -81,6 +102,19 @@ def _calculate_bm25(self, tokenized_document: list[str]) -> Vector:
         Returns:
             Vector: BM25 vector for document.
         """
+        if not tokenized_document or not isinstance(tokenized_document, list) \
+            or not all(isinstance(token, str) for token in tokenized_document):
+            raise ValueError
+
+        vector = [0.0 for _ in self._vocabulary]
+        bm25 = calculate_bm25(self._vocabulary, tokenized_document,
+                              self._idf_values,
+                              avg_doc_len=self._avg_doc_len, doc_len=len(tokenized_document))
+        if not bm25:
+            return()
```

Comment:
Is return a function? Do you call return? what is it? Does it work?

Reply:
Forgot to change this.... but it worked

Reply:
Done?

Reply:
Done